### PR TITLE
Refactor rewrite middleware. Use rack-rewrite.

### PR DESCRIPTION
**Pull request by nateberkopec**
I changed the logic very slightly on the first one, let me know if that's out of line.

### CODE REVIEW

**Comment by nateberkopec** [comment]
Trying to think of a way to make the rewrite code reusable so we can keep the tests, but [`rack-rewrite` depends on `block_given?` when initializing](https://github.com/jtrupiano/rack-rewrite/blob/master/lib/rack/rewrite.rb#L15).

File: test/unit/redirector_test.rb [hunk 1]
```diff
@@ -1,91 +0,0 @@
-require 'test_helper'
```
Comment (arthurnn) [inline test/unit/redirector_test.rb]: should we keep the tests to make sure we did the parameterization of the gem correctly? 
I dont think we should test the gems functionality, however I am also not sure how it works, and our code was working pretty fine before and it was well tested.

Comment (nateberkopec) [inline test/unit/redirector_test.rb]: Yeah I'll think of a clean way to keep these

**Comment by kbrock** [comment]
this looks nice. thanks

**Pull request closed without merging**